src/main/scala/mesosphere/marathon/state/PathId.scala

```diff
@@ -231,27 +218,28 @@ object PathId {
     * For external usage. Needed to overwrite the whole description, e.g. id.path -> id.
     */
   implicit val pathIdValidator = validator[PathId] { path =>
-    path is childOf(path.parent)
     path is validPathChars
     path is withoutReservedKeywords
+    path is childOf(path.parent)
   }
 
   /**
     * Validate path with regards to some parent path.
     * @param base Path of parent.
     */
   def validPathWithBase(base: PathId): Validator[PathId] = validator[PathId] { path =>
-    path is childOf(base)
     path is validPathChars
+    path is childOf(base)
   }
 
   /**
     * Make sure that the given path is a child of the defined parent path.
     * Every relative path can be ignored.
     */
   private def childOf(parent: PathId): Validator[PathId] = {
-    isTrue[PathId](s"Identifier is not child of $parent. Hint: use relative paths.") { child =>
-      !parent.absolute || (child.canonicalPath(parent.asAbsolutePath).parent == parent)
+    isTrue[PathId](s"Identifier is not child of '$parent'") {
+      case _: RelativePathId => true
```

Comment:
It's the parent on the left `!parent.absolute` but child `case _: RelativePathId` on the right. This is at least not identical.

Reply:
```
isTrue[PathId](s"Identifier is not child of '$parent'") { child =>
      match parent {
        case _: RelativePathId => true
        case p: AbsolutePathId => child.canonicalPath(p).parent == parent
      }
    }
```

This looks better, right?